internal/fwserver/server_updateresource.go

```diff
@@ -91,6 +93,32 @@ func (s *Server) UpdateResource(ctx context.Context, req *UpdateResourceRequest,
 		updateReq.ProviderMeta = *req.ProviderMeta
 	}
 
+	if req.PlannedPrivate != nil {
+		updateReq.Private = req.PlannedPrivate.Provider
+		updateResp.Private = req.PlannedPrivate.Provider
+		resp.Private = req.PlannedPrivate
+	}
+
```

Comment:
I think this logic is extraneous, given the below logic

```suggestion
```

Reply:
Good catch! I meant to remove that.